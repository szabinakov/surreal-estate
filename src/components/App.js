import React, { useState } from 'react';
import '../styles/App.css';
import { Switch, Route } from 'react-router-dom'

import NavBar from './NavBar'
import Properties from './Properties'
import AddProperty from './AddProperty'

function App() {
  const [userID, setUserID] = useState('')

  const handleLogin = (response) => {
    setUserID(response.userID)
  }

  const handleLogout = () => {
    window.FB.logout((response) => setUserID(''))}

  return (
    <>
      <NavBar userID={userID} onLogin={handleLogin} onLogout={handleLogout}/>
      <Switch>
        <Route exact path='/' component={Properties}/>
        <Route path='/add-property' component={AddProperty}/>
      </Switch>
    </>
  );
}

export default App;
