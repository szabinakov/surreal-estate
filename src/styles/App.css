body {
    background-color: rgba(158, 158, 158, 0.144);
    font-family: 'Nunito Sans', sans-serif;
    padding: 0;
    margin: 0;
}