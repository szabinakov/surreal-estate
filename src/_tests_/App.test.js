import React from 'react';
import { render } from '@testing-library/react';
import App from '../components/App';
import { BrowserRouter } from 'react-router-dom'

describe('App', () => {
  it('renders correctly', ()=> {
    const {asFragment} = render (
      <BrowserRouter>
      <App/>
      </BrowserRouter>
      )

    expect(asFragment).toMatchSnapshot()
  })})
