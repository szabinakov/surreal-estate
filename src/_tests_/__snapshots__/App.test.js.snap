// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <img
      alt="Website Logo"
      class="logo"
      src="logo.png"
    />
    <ul
      class="navbar-links"
    >
      <li>
        <a
          class="navbar-links-item"
          href="/"
        >
          View Properties
        </a>
      </li>
      <li>
        <a
          class="navbar-links-item"
          href="/add-property"
        >
          Add Property
        </a>
      </li>
    </ul>
  </div>
  <h2>
    Properties
  </h2>
  <div
    class="SideBar"
  >
    <form
      class="SearchForm"
    >
      <input
        class="SearchInput"
        placeholder="Search!"
        type="text"
        value=""
      />
      <button
        class="SearchButton"
        type="submit"
      >
        Search!
      </button>
    </form>
    <p>
      <a
        href="/?query={\\"city\\":\\"Manchester\\"}"
      >
        Manchester
      </a>
    </p>
    <p>
      <a
        href="/?query={\\"city\\":\\"Leeds\\"}"
      >
        Leeds
      </a>
    </p>
    <p>
      <a
        href="/?query={\\"city\\":\\"Sheffield\\"}"
      >
        Sheffield
      </a>
    </p>
    <p>
      <a
        href="/?query={\\"city\\":\\"Liverpool\\"}"
      >
        Liverpool
      </a>
    </p>
    <p>
      <a
        href="/?sort={\\"price\\":1}"
      >
        Price Ascending
      </a>
    </p>
    <p>
      <a
        href="/?sort={\\"price\\":-1}"
      >
        Price Descending
      </a>
    </p>
  </div>
  <div
    class="PropertyContainer"
  />
</DocumentFragment>
`;
